sdk/core/src/main/java/com/klaviyo/coresdk/Klaviyo.kt

```diff
@@ -1,73 +1,140 @@
+     * (e.g. after a fresh login)
+     *
+     * @param phoneNumber Phone number for active user
+     */
+    fun setPhoneNumber(phoneNumber: String): Klaviyo = apply {
+        this.setProfile(Profile().setPhoneNumber(phoneNumber))
+    }
+
+    /**
+     * Assigns an external ID to the current internally tracked profile
+     *
+     * The SDK keeps track of current profile details to
+     * build analytics requests with profile identifiers
+     *
+     * This should be called whenever the active user in your app changes
+     * (e.g. after a fresh login)
      *
-     * @param customerProperties A map of customer property information.
-     * Defines the customer that triggered this event
-     * @param properties A map of event property information.
-     * Additional properties associated to the event that are not for identifying the customer
+     * @param id Phone number for active user
+     * @return
      */
-    fun track(
-        event: KlaviyoEvent,
-        customerProperties: KlaviyoCustomerProperties,
-        properties: KlaviyoEventProperties? = null
-    ) {
-        val request = TrackRequest(KlaviyoConfig.apiKey, event, customerProperties, properties)
-        processRequest(request)
+    fun setExternalId(id: String): Klaviyo = apply {
+        this.setProfile(Profile().setIdentifier(id))
     }
 
     /**
-     * Creates an identify request for the Klaviyo APIs and begins processing.
+     * Assign arbitrary attributes to the current profile by key
+     *
+     * This should be called when you collect additional data about your user
+     * (e.g. first and last name, or an address)
+     *
+     * @param propertyKey
+     * @param value
+     * @return
+     */
+    fun setProfileAttribute(propertyKey: KlaviyoProfileAttributeKey, value: String): Klaviyo =
+        setProfile(Profile().also { it[propertyKey] = value })
+
+    /**
+     * Queues a request to identify profile properties to the Klaviyo API
+     *
+     * Any new identifying properties (externalId, email, phone) will be saved to the current
+     * internally tracked profile. Otherwise any identifiers missing from [profile] will be
+     * populated from the current internally tracked profile info.
+     *
      * Identify requests track specific properties about a user without triggering an event
      *
-     * @param properties A map of properties that define the user
+     * @param profile A map of properties that define the user
+     * @return
      */
-    fun identify(properties: KlaviyoCustomerProperties) {
-        val request = IdentifyRequest(KlaviyoConfig.apiKey, properties)
-        processRequest(request)
+    fun setProfile(profile: Profile): Klaviyo = apply {
+        // TODO debounce so fluent setters don't have to create 1 request per call
+        // Note - is there value to tighter coupling the UserInfo update with the API call
+        UserInfo.mergeProfile(profile)
+        Registry.apiClient.enqueueProfile(profile)
```

Comment:
shouldnt this enqueue `UserInfo.getAsProfile()`?

Reply:
`UserInfo` is only concerned with identifiers, so `getAsProfile` would return a `Profile` object with *just* the identifiers. 

`mergeProfile` adds identifiers from internal state to the attributes of the `Profile` object before we enqueue.

Reply:
but if there are identifiers that are added because of the merge, API request should include both right?

Reply:
oh wait so youre saing that `mergeProfile` changes the state of the provided `profile` input?

Reply:
Yes, and thats what `mergeProfile` does, it mutates the `profile` arg.
The naming may be causing confusion, let me think if there's a better name for `mergeProfile`

Reply:
is there a way we can avoid mutation?

Reply:
generally thats ripe for bug infestations

Reply:
Perhaps that would be best... I could make mergeProfile return `getAsProfile` instead, so it is a new profile object.

Reply:
Resolving, I made some changes here including to add the debouncing but also took out the merge/mutation thing.